src/main/java/net/earthcomputer/clientcommands/command/TicTacToeCommand.java

```diff
@@ -0,0 +1,280 @@
+                    }
+
+                    Minecraft.getInstance().gui.getChat().addMessage(Component.translatable("c2cpacket.startTicTacToeGameC2CPacket.outgoing.accept"));
+                })))
+                .withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, Component.translatable("c2cpacket.startTicTacToeGameC2CPacket.incoming.accept.hover")))))
+            .append("]");
+        Minecraft.getInstance().gui.getChat().addMessage(component);
+    }
+
+    private static int open(FabricClientCommandSource source, String name) throws CommandSyntaxException {
+        TicTacToeGame game = activeGames.get(name);
+        if (game == null) {
+            throw NO_GAME_WITH_PLAYER_EXCEPTION.create();
+        }
+
+        source.getClient().tell(() -> source.getClient().setScreen(new TicTacToeGameScreen(game)));
+        return Command.SINGLE_SUCCESS;
+    }
+
+    public static void onPutTicTacToeMarkC2CPacket(PutTicTacToeMarkC2CPacket packet) {
+        String sender = packet.sender();
+        TicTacToeGame game = activeGames.get(sender);
+        if (game == null) {
+            return;
+        }
+        if (game.putMark(packet.x(), packet.y(), game.yourMarks.opposite())) {
+            if (game.getWinner() == game.yourMarks.opposite()) {
+                Minecraft.getInstance().gui.getChat().addMessage(Component.translatable("c2cpacket.putTicTacToeMarkC2CPacket.incoming.lost", sender));
+                activeGames.remove(sender);
+                return;
+            }
+            MutableComponent component = Component.translatable("c2cpacket.putTicTacToeMarkC2CPacket.incoming", sender);
+            component.withStyle(style -> style
+                .withClickEvent(new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/ctictactoe open " + sender))
+                .withHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, Component.literal("/ctictactoe open " + sender))));
+            Minecraft.getInstance().gui.getChat().addMessage(component);
+        }
+    }
+
+    private static class TicTacToeGame {
+        public final PlayerInfo opponent;
+
+        private final Mark[][] board = new Mark[3][3];
+        private final Mark yourMarks;
+        private boolean yourTurn;
+
+        public TicTacToeGame(PlayerInfo opponent, Mark yourMarks) {
+            this.opponent = opponent;
+            this.yourMarks = yourMarks;
+            this.yourTurn = yourMarks == Mark.CROSS;
+        }
+
+        public boolean putMark(byte x, byte y, Mark mark) {
+            if (this.yourMarks == mark == this.yourTurn) {
+                if (this.board[x][y] == null) {
+                    this.board[x][y] = mark;
+                    this.yourTurn = !this.yourTurn;
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        public Mark getWinner() {
+            for (byte x = 0; x < 3; x++) {
```

Comment:
Why is byte used here instead of int?

Reply:
Idk because it could be 😄. It's also a byte in the packet.